examples/querying_entity_traits.ipynb

```diff
@@ -0,0 +1,710 @@
+    "\n",
+    "The following examples illustrate usage of the `entityTraits` method."
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "d8eaf0a116f06949"
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4348da46-9467-4734-9f96-d84e26f7c11f",
+   "metadata": {
+    "editable": true,
+    "jp-MarkdownHeadingCollapsed": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "## Querying the full set of traits associated with an entity"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5e32a587-5233-441e-9b80-1b2d6d3769fb",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "OpenAssetIO does not try to fully define entity types. What it does define are traits that an entity satisfies. An entity is theoretically fully specified by some (large) subset of traits in the universe of all possible traits. A manager understands some subset of that theoretical complete subset of traits. This is the \"trait set\" of the entity, with respect to a given manager.\n",
+    "\n",
+    "In order to find out what the trait set of an entity is, we can use the `entityTraits` API method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0fcf0214-2561-488a-8eee-0330ca27710d",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "entity_trait_set = manager.entityTraits(logo_ref, EntityTraitAccess.kRead, context)\n",
+    "\n",
+    "helpers.display_result(entity_trait_set)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a6c4f0ff-0e5d-40b7-84d6-d46efa8cba7a",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "`EntityTraitAccess.kRead` tells the manager the reason why you want the trait set. In this case we want to use the result in a read context, e.g. `resolve`ing properties, grouping entities of the same type in a UI, or determining the type of entity after a drag-and-drop of an entity reference into the application.\n",
+    "\n",
+    "What happens when we `resolve` this trait set?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "entity_data = manager.resolve(logo_ref, entity_trait_set, ResolveAccess.kRead, context)\n",
+    "\n",
+    "helpers.display_result(entity_data)"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "fbae28c950c33589"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Note that not all traits are associated with `resolve`able properties - many are simply tags. However, it is safe to `resolve` the full trait set - those that do not have properties are simply missing from the response.\n",
+    "\n",
+    "As an aside, we can filter the trait set through `managementPolicy`"
```

Comment:
Might be nice to call the motivation out a little earlier.
```suggestion
    "As an aside, we can filter the trait set through `managementPolicy` if we wish to know in advance, which traits any given manager is capable of providing data for"
```

Reply:
Updated as suggested in ae164b24